src/vivarium_csu_zenon/components/observers.py

```diff
@@ -208,9 +210,7 @@ def get_state_person_time(pop, config, disease, state, current_year, step_size,
 
 
 def get_transition_count(pop, config, disease, transition, event_time, age_bins):
-    from_state, to_state = transition.split('_TO_')
-    event_this_step = ((pop[f'{to_state}_event_time'] == event_time)
-                       & (pop[f'previous_{disease}'] == from_state))
+    event_this_step = pop[disease] != pop[f'previous_{disease}']
```

Comment:
@KjellSwedin You'll want to bring this pattern over to lsff to work with iron and vitamin a deficiency